### ISSUE TITLE
Click handler is not executed for top level menu items

### ISSUE THREAD

**Issue by petrfelzmann** [electron/electron#3049]
I create menu in render process.  But the `console.log` below is not executed for top level menu items.

``` javascript
var remote = require('remote'),
    Menu = remote.require('menu'),
    MenuItem = remote.require('menu-item');

var tmp = [{
    label: 'menu item',
    click: function() {
        console.log('menu item clicked'); // not executed
    }
}];

var menu = Menu.buildFromTemplate(tmp);
Menu.setApplicationMenu(menu);
```

**Comment by etiktin** [comment]
Does it work when you execute this on the main process and not the renderer (on Windows you'll need to run the app with `--enable-logging` to see the output)?

**Comment by petrfelzmann** [comment]
No, it's not working even from main process.

**Comment by hannanabdul55** [comment]
Hey, I was wondering if I can start contributing to this project by fixing this bug.

**Comment by etiktin** [comment]
Sure. Contributions are welcomed.

**Comment by fdelayen** [comment]
It works fine if you follow the [doc](https://github.com/electron/electron/blob/master/docs/api/remote.md).
According to this one, it is not like a context menu. Your menu should have a submenu like that : 

```javascript
let menu = [{
  label: 'Menu',
  role: 'menu',
  submenu: [{
    label: 'menu item',
    click: function() {
      console.log('menu item clicked'); // executed
    }
  }]
}];
```
For the moment your code is creating an "Activate" submenu automatically and the click does not make it to the javascript side ([menu-item.js](https://github.com/electron/electron/blob/master/lib/browser/api/menu-item.js)) but I don't even know if it should appear.

@zcbenz What do you think about it ? It can search deeper why it does not work or not create top level menu without submenu. I don't know what is the most consistent solution.

**Comment by fdelayen** [comment]
At least, that's what it's doing on Linux (and dbus is not notifying us that it's been activated, that's really strange). On windows, it doesn't create that strange submenu, but the click does not work either.
As menus are defined separately (globalmenu via dbus for X11 and another class for Windows), I'm not sure whether these bugs are related or platform-per-platform.
I'm gonna debug that on Windows to see what's happening when I'll have a bit of time :stuck_out_tongue:

**Comment by tonyganch** [comment]
Hi, I wanted to check if the issue is fixed already since I see there is a PR merged.

**Comment by codebytere** [comment]
Closing due to abandonment, but yes @tonyganch i think it has

**Comment by IshanArya** [comment]
The problem still exists for MacOS at least.

**Comment by DesselBane** [comment]
It still exists on Linux too

**Comment by asmab** [comment]
Same problem on Linux

**Comment by codebytere** [comment]
@DesselBane @IshanArya there is another related issue open at https://github.com/electron/electron/issues/8703, track that one for ongoing efforts to address this

**Comment by tHBp** [comment]
Unfortunately, the issue linked above is closed as duplicate of #2895, which itself has been abandoned. The issue is fixed for Windows but affects macOS still (can't say for Linux though).

**Comment by MarkNBroadhead** [comment]
This works as expected with MacOS 11.1 Big Sur, Node 12.18.3, Chrome 87.0.4280.88, and Electron 11.1.1.
